### PR TITLE
wip: show OSM on search pages that use a location

### LINKED ISSUE #591: Add support for OpenStreetMap and Mapbox
Maps are shown using Google Maps. Firstly Google maps has nice features, but most good features now are very costly. Secondly Google maps is a proprietary map that collects data of our users.

Therefore please implement OpenStreetMap and Mapbox 
- [ ] for user maps in Wizard step 1 
- [x] show OSM/mapbox for locations on public event pages
- [ ] show OSM/mapbox on search pages that use a location
- [ ] provide an option to add a key for the service in the admin page at https://eventyay.com/admin/settings

As part of this issue 
* please take out the map code from the project and ensure there are no console errors.
* If required make necessary changes on the server.

Related to https://github.com/fossasia/open-event-frontend/issues/3564

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes #591

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by maze-runnar** [comment]
@iamareebjamal  this should be a static map, that shows location which is entered in search box, or `draggable` that is also used to set location?  And how to get latitude and longitude from `locationName`

File: app/components/explore/side-bar.js [hunk 1]
```diff
@@ -16,6 +16,8 @@ export default class SideBar extends Component {
   customEndDate = null;
   @tracked showFilters = false;
   isMapVisible = true;
+  lat = 27.00;
+  lng = 79.00;
```
Comment (iamareebjamal) [inline app/components/explore/side-bar.js:20]: But this is static so what's the point

File: app/templates/components/explore/side-bar.hbs [hunk 2]
```diff
@@ -1,3 +1,11 @@
+<LeafletMap @lat={{this.lat}} @lng={{this.lng}} @zoom={{15}} as |layers|>
+  <layers.tile @url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png" @attribution="&copy; <a href='https://www.openstreetmap.org/copyright'>OpenStreetMap</a> contributors" />
+  <layers.marker @location={{array this.lat this.lng}} as |marker|>
+    <marker.popup>
+      {{this.location}}
+    </marker.popup>
+  </layers.marker>
+</LeafletMap>
```
Comment (iamareebjamal) [inline app/templates/components/explore/side-bar.hbs:8]: Duplication from EventMap. Extract to remove duplicate code

File: app/components/explore/side-bar.js (hunk 1 above)
Comment (maze-runnar) [reply app/components/explore/side-bar.js:20]: it's just initial, means I will change it.

**Comment by iamareebjamal** [comment]
It's just OSM, mapbox will be added later

File: app/templates/components/explore/side-bar.hbs (hunk 2 above)
Comment (maze-runnar) [reply app/templates/components/explore/side-bar.hbs:8]: > It's just OSM, mapbox will be added later

ok

**Comment by iamareebjamal** [comment]
Even if it is draggable, we don't support coordinate search for now. You don't extract coordinates from locationName, you do that from event.latitude and event.longitude

**Comment by maze-runnar** [comment]
> Even if it is draggable, we don't support coordinate search for now. You don't extract coordinates from locationName, you do that from event.latitude and event.longitude

Can i get the event.latitude from the searched events by location,  from the first event AND show it on map , while search by location triggers?

**Comment by iamareebjamal** [comment]
https://nominatim.openstreetmap.org/search?q=lifelong+institute&format=jsonv2&addressdetails=1
https://nominatim.openstreetmap.org/reverse?lat=1.3196263&lon=103.8928117&format=jsonv2

**Comment by iamareebjamal** [comment]
If you can do draggable and reverse geocoding, it'll work

**Comment by iamareebjamal** [comment]
So, if someone enters a location, you do search and set lat lng to first result in nominatim search. And when someone clicks on a point, you get lat lng and you do reverse search and set the location to first result. You'll have to be less specific in setting location name, if you set it exactly, like "Lifelong Learning Institute, Eunos Road 8, Geylang, Southeast, 409051, Singapore", the site search will return no results

**Comment by maze-runnar** [comment]
how to get value from infinity model like `this.model.filteredEvents`. this.model.filteredEvents[0].latitude is giving `latitude of undefined error`.

**Comment by iamareebjamal** [comment]
You don't need latitude from that. You need latitude from the API I linked above

**Comment by maze-runnar** [comment]
> You don't need latitude from that. You need latitude from the API I linked above

ok

**Comment by maze-runnar** [comment]
@iamareebjamal I have made changes. But it is not changing position on map on searching, it needs a page refresh. What to do next?

**Comment by iamareebjamal** [comment]
@maze-runnar Please take up another issue. This is part of the freelance project and we don't want any overlap

**Comment by maze-runnar** [comment]
the issue was very near to be solved but the issue has got a `project` label, so that is part of freelancing and this PR doesn't matter now.

**Pull request closed without merging**